docs/userguide/icu4j/index.md

```diff
@@ -68,41 +67,122 @@ There are a few different ways to download the ICU4J releases.
+            <version>68.1</version>
+        </dependency>
+
+        <dependency>
+            <groupId>com.ibm.icu</groupId>
+            <artifactId>icu4j-charset</artifactId>
+            <version>68.1</version>
+        </dependency>
+
+        <dependency>
+            <groupId>com.ibm.icu</groupId>
+            <artifactId>icu4j-localespi</artifactId>
+            <version>68.1</version>
+        </dependency>
+        ```
 
 *   **GitHub Source Repository:**
-    If you are interested in developing features, patches, or bug fixes for ICU4J, you should probably be working with the latest version of the ICU4J source code. You will need to clone and checkout the code from our GitHub repository to ensure that you have the most recent version of all of the files. There are several ways to do this. Please follow the directions that are contained on the [Source Repository page](https://icu.unicode.org/repository) for details.
+    If you are interested in developing features, patches, or bug fixes for ICU4J, you should probably be working with the latest version of the ICU4J source code. You will need to clone and checkout the code from our GitHub repository to ensure that you have the most recent version of all of the files. There are several ways to do this. Please follow the directions that are contained on the [Source Code Setup page](../../devsetup/source/) for details.
 
 For more details on how to download ICU4J directly from the web site, please see the ICU download page at [https://icu.unicode.org/download](https://icu.unicode.org/download)
 
-## The Structure and Contents of ICU4J
+
+## The Structure and Contents of ICU4J - ICU 74 and later
+
+Below, all directory paths are relative to the directory where the ICU4J source archive is extracted.
+
+### Information and build files
+
+| Path                         | Description                                                                             |
+|------------------------------|-----------------------------------------------------------------------------------------|
+| `pom.xml`                   | The root Maven build file for ICU4J. See [How to Install and Build](#how-to-install-and-build---icu-74-and-later) for more information |
+| `LICENSE` | ICU license                                                                             |
+
+### ICU4J runtime class files
+
+Each sub-component is represented in the Maven build as a separate Maven module / project.
+
+The directory structure of the codebase within each module follows Maven defaults.
+Thus, runtime class sources will be in the `src/main/java` subdirectory of the module.
+Ex: the class `com.ibm.icu.text.Collator` can be found at `<ICU>/icu4j/main/collate/src/main/java/com/ibm/icu/text/Collator.java`.
+Any resource files (non-source files) will be located at `src/main/resources`.
+
+
+| Sub-component Path                    | Sub-component Name | Build Dependencies        | Public API Packages | Description   |
```

Comment:
I wonder if it is worth listing this.
Or at least the dependency, and component names.
These are things that can be easily obtained with maven:
```
mvn dependency:tree
```

Reply:
I included that to be consistent with the Ant-based docs we have for ICU 73 and earlier. The same argument could have been made there ("just look at the build.xml file to find component deps").

Personally, I think documentation _about a codebase_ is something that I find useful and would like to see more of, even though it is rare. But I do believe it to be a different type of doc than inline source comments, API docs, design/architecture docs, and tutorials.

Reply:
Yes, but documentation about code base also gets outdated very fast :-)

For the Ant dependencies you had to grep, then "massage" things yourself.
Maven has a standard way to do it.

What about keep the table, but remove the "Dependencies" column, and say "for an up to date dependency report run `mvn dependency:tree`"?
The result is more readable than this table (for example you can see if something is a test dependency or not)